### ISSUE TITLE
[IBEX_GUI]: Tabbing out from script generator fields [Timebox: 4.5D]

### ISSUE THREAD

**Issue by Chsudeepta** [ISISComputingGroup/IBEX#8122]
## Where?
As a user of script generator, if I load a enum based script and then add a row, select a value and tab out from the field the GUI starts misbehaving.

## How?
1. Open IBEX GUI
2. Select Script Generator from left hand menu
3. Select a enum based script from Script Definition drop-down in the script composer widget.
4. Add a new row using the buttons in the right hand side.
5. Select a value for Process Status from the drop down and press TAB.
6. The GUI becomes unresponsive and lots of exceptions are raised (Check the console logs).

## Reproducible?
[Yes]

### To Reproduce
- 

## Acceptance criteria 
- [ ] Tabbing out of the field should take you to the next field in the tablist. Check functionality in other script types
- [ ] The code should follow standard coding practices
- [ ] There should be no issues in building the code using the designated batch file, including no test failures and checkstyle failures.

## How to Test 
_verbose instructions for reviewer to test changes
(**Add before making a PR**)_

**Comment by Chsudeepta** [comment]
I have analysed the issue and found that the root cause is that in the callbacks for add/insert/delete rows there is a get on an invalid index (value 1, java index starts from 0) in ScriptGeneratorViewModel. This is used to check whether the added action parameters corresponds to an enum script. I found that the enum script has only action parameter and if we really want to add this check we can either use the index 0, or we can surround the check with an upper check on the number of action parameters available in the first place. 
		if (scriptGeneratorModel.getActionParameters().size() > ActionsViewTable.NON_EDITABLE_COLUMNS_ON_LEFT) {
			/** Rest of the code comes here **/
		}
